src/pymap/settings.py

```
@@ -213,85 +226,101 @@
-        with open(".secret", "r", encoding="utf-8") as fh:
-            read_key: str = fh.read().strip()
-            if len(read_key) > 10:
-                SECRET_KEY = read_key
-            else:
-                raise ValueError("The loaded SECRET_KEY is too short.")
-    except (OSError, FileNotFoundError, ValueError) as e:
-        print(f"Failed to load secret file: {e}")
-        sys.exit(1)
+        secret_key = secret_file.read_text(encoding="utf-8").strip()
+    except OSError as e:
+        raise ImproperlyConfigured(f"Error reading secret key file: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)